Sources/OpenAPIKit/Utility/OrderedDictionry+LocallyDereferenceable.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  OrderedDictionry+LocallyDereferenceable.swift
+//  
+//
+//  Created by Alberto Lagos on 11-09-23.
+//
+
+import OpenAPIKitCore
+import Foundation
+
+/// A Swift extension for dereferencing an `OrderedDictionary` conforming to the `LocallyDereferenceable` protocol.
+///
+/// - Parameters:
+///   - components: The OpenAPI components containing definitions.
+///   - references: A set of references to track dereferenced items.
+///   - name: The name of the component from which the dereferencing is initiated.
+///
+/// - Returns: A dereferenced `OrderedDictionary` containing keys and values of dereferenced types.
+///
+/// - Throws: An error if dereferencing fails for any element.
+extension OrderedDictionary: LocallyDereferenceable where Key: LocallyDereferenceable, Key.DereferencedSelf: Hashable, Value: LocallyDereferenceable {
+
+    public func _dereferenced(in components: OpenAPI.Components,
+                              following references: Set<AnyHashable>,
+                              dereferencedFromComponentNamed name: String?) throws -> OpenAPIKitCore.OrderedDictionary<Key.DereferencedSelf, Value.DereferencedSelf> {
+
+        try reduce(into: OrderedDictionary<Key.DereferencedSelf, Value.DereferencedSelf>()) { result, element in
+            let key = try element.key._dereferenced(in: components, following: references, dereferencedFromComponentNamed: name)
```

Comment:
This is all looking great! The only thing is a bit of a subtle point that I probably don't have documented very well if at all: This (and the other locations you've added) should be passing `dereferencedFromComponentName: nil`. This parameter was meant to only be set when called from a `JSONReference` (or `OpenAPI.Reference`) but then used by anything that has `vendorExtensions` to store an `x-component-name` extension.

For example, it is passed from https://github.com/mattpolzin/OpenAPIKit/blob/release/3_0/Sources/OpenAPIKit/JSONReference.swift#L534..L536.

And it is used by https://github.com/mattpolzin/OpenAPIKit/blob/release/3_0/Sources/OpenAPIKit/Path%20Item/DereferencedPathItem.swift#L67..L70.

But anywhere else that calls` _dereferenced(in...)` passes `nil` like https://github.com/mattpolzin/OpenAPIKit/blob/release/3_0/Sources/OpenAPIKit/Path%20Item/DereferencedPathItem.swift#L55..L55